Sources/Workspace/Workspace.swift

```diff
@@ -752,6 +813,49 @@ extension Workspace {
         }
     }
 
+    /// Loads and returns manifest at the given path.
+    public func loadRootManifest(
+        at path: AbsolutePath,
+        diagnostics: DiagnosticsEngine,
+        completion: @escaping(Result<Manifest, Error>) -> Void
+    ) {
+        self.loadRootManifests(packages: [path], diagnostics: diagnostics) { result in
+            completion(result.tryMap{
+                // normally, we call loadRootManifests which attempts to load any manifest it can and report errors via diagnostics
+                // in this case, we want to load a specific manifest, so if the diagnostics contains an error we want to throw
+                guard !diagnostics.hasErrors else {
+                    // not sure about this one
+                    throw StringError("\(diagnostics)")
```

Comment:
I think there is a `fatalError` for cases in which diagnostics have already been reported.